include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp

```diff
@@ -104,12 +106,15 @@ class score_matrix_single_column
      */
     template <std::integral column_index_t, std::integral row_index_t>
     void resize(column_index_type<column_index_t> const number_of_columns,
-                row_index_type<row_index_t> const number_of_rows)
+                row_index_type<row_index_t> const number_of_rows,
+                score_t const initial_value = score_t{})
     {
         this->number_of_columns = number_of_columns.get();
-        optimal_column.resize(number_of_rows.get());
-        horizontal_column.resize(number_of_rows.get());
-        vertical_column = views::repeat_n(score_t{}, number_of_rows.get());
+        optimal_column.clear();
+        horizontal_column.clear();
```

Comment:
yes, because resize only initiliases the elements that are added. But we want the "entire" matrix to be re-filled with the init value. That is if the band parameter changes in between calls of the alignment, which is not the case right now but might be in the future, the matrix might not be properly initiliased.